### DISCUSSION TITLE
Assign classification store collections to data objects

### DISCUSSION THREAD

**Discussion by cmann90** [pimcore/pimcore#10436] [Q&A / Support]
I am trying to assign classification store collections to new data objects.
Everything works fine when I want to assign groups but when I try to add collections through code the Collection ID is missing in the database.

Any idea what I am doing wrong?
Is there a reason why no "setActiveCollections" method is provided?

Thanks for any help/guidance in advance :-)

**My Code**
```php
$activeGroups = array(
    1565 => true
);

$groupCollectionMapping = array(
    1565 => 1
);

$product->getPiBasics()->setActiveGroups($activeGroups);
$product->getPiBasics()->setGroupCollectionMappings($groupCollectionMapping);

$product->setOmitMandatoryCheck(true);
$product->save();
```

**UI looks fine**
![image](https://user-images.githubusercontent.com/68635243/135065486-1a6091a9-a026-4a9f-95c5-11475fc93dca.png)

**collectionId = NULL in db table**
![image](https://user-images.githubusercontent.com/68635243/135065503-0beed6cb-fb58-47ee-852b-87fe417eb3fb.png)

**When the collection is added through the UI the db is as it is supposed to be**
![image](https://user-images.githubusercontent.com/68635243/135067052-77271bad-e27d-4de2-a2b8-4bf14c942756.png)

**Documentation which I used as reference:**
https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Classification_Store.html

**Comment by frithjof** [comment]
the collection id is only saved with a field value ... so you need to at least save one entry in the group
```

$activeGroups = array(
    1565 => true
);

$groupCollectionMapping = array(
    1565 => 1
);

$product->getPiBasics()->setActiveGroups($activeGroups);
$product->getPiBasics()->setGroupCollectionMappings($groupCollectionMapping);
$product->getPiBasics()->setLocalizedKeyValue(1565 , $somekeyid,'testdata');
$product->setOmitMandatoryCheck(true);
$product->save();
```

**Reply by cmann90** [reply]
Frithjofs answer works but doesn't feel right as I would have to insert data to correctly save the collection id.
What I wanted to accomplish was to add a collection without inserting data. Now I also noticed that collections do not behave like groups.

**Behavior I expected from collections**
1. Add collections to data objects
2. Change collection group associations
3. All data objects are updated

**Actual behavior**
1. Add collections to data objects
2. Change collection group associations
3. Existing data objects are not updated


Groups behave this way. Once you add or remove a key from a group this change is reflected in all data objects the group is assigned to.

So I am wondering why collections behave differently from groups and especially why the collection id is saved in tables **object_classificationstore_data_\<data object id\>**? 
When I add a collection on the UI and add data to one group but not another then the collection id will only be saved for the group with keys that contain data... why is the collection id stored in this table when the saved information is not reliable?

Data inconsistencies which can occur
- Groups which where added in the admin UI through collections could be missing the id in the database table when the data object is saved without adding key data. After a reload of the data object the collection id won't be saved 
- Adding or removing groups from collections is not updated in data objects therefore the saved collection id in **object_classificationstore_data_\<data object id\>** does not reflect the current definition of the collection.

Maybe I am missing something? Hopefully someone from Pimcore can shine some light upon this.
@fashxp
